### PR TITLE
Forgot password

### PR DESCRIPTION
**Pull request by sagarsubedi**
- [x] Forgot password button on the login page
- [x] Store question and answer in database
- [x] Functionality to change password
- [x] Create security questions during the signup process
- [x] Add new password reset page
- [x] Allow users to change security Q&A

1. All new users will be added to enter a security question and answer during the signup process.
2. User can change their security question and answer when they are logged in from the login page.
3. In case the user forgot their password, they can create a new one by answering their security question.

All tasks completed. Code tested.

### CODE REVIEW

File: auth.py [hunk 1]
```diff
@@ -109,7 +116,7 @@ def register():
                 else:
                     # if new user, then hash the password, insert to table and log them in
                     hashed_password = generate_password_hash(password1, method='sha256')
-                    user = dict(email=email, password=hashed_password)
+                    user = dict(email=email, password=hashed_password, view='calendar', security_question=security_question, security_answer=security_answer)
```
Comment (Qtipshooter) [inline auth.py:119]: We are no longer storing the view in the user table as it makes more sense to store it in the customization table as far as I know.

File: static/styles/styles.css [hunk 2]
```diff
@@ -225,4 +225,51 @@ body{
     /* When the radio button is checked, add a yellow background */
     .radio input:checked ~ .checkmark {
       background-color: var(--primary-yellow);
-    }
\ No newline at end of file
+    }
+
+/* for forgot password page */
```
Comment (Qtipshooter) [inline static/styles/styles.css:230]: If these are only being used on this page, do you want to move them into a new file that is just for that page?

**Comment by Qtipshooter** [comment]
Looks really nice code wise, I will look at the Site itself from your branch when I get the chance and play with it

File: auth.py (hunk 1 above)
Comment (sagarsubedi) [reply auth.py:119]: I will remove that view field from there.

File: static/styles/styles.css (hunk 2 above)
Comment (sagarsubedi) [reply static/styles/styles.css:230]: Yeah, I can do that

**Review by Roodude15 — APPROVED** [review]
Everything looks good.

**Review by Qtipshooter — APPROVED** [review]

**Pull request merged**